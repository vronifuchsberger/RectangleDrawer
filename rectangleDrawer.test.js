const drawRectangle = require("./rectangleDrawer");

test("draws a 2 by 2 rectangle", () => {
  expect("\n" + drawRectangle(2, 2)).toMatchSnapshot();
});

test("draws a 5 by 5 rectangle", () => {
  expect("\n" + drawRectangle(5, 5)).toMatchSnapshot();
});

test("draws a 5 by 3 rectangle", () => {
  expect("\n" + drawRectangle(5, 3)).toMatchSnapshot();
});

test("minimum size", () => {
  expect(() => drawRectangle(1, 2)).toThrow();
  expect(() => drawRectangle(3, 1)).toThrow();
});

test("infinite input", () => {
  expect(() => drawRectangle(Infinity, 3)).toThrow();
});

test("non numerical input", () => {
  expect(() => drawRectangle()).toThrow();
  expect(() => drawRectangle("test", 3)).toThrow();
  expect(() => drawRectangle(true, 3)).toThrow();
});

test("non integer input", () => {
  expect(() => drawRectangle(5.1, 3)).toThrow();
});
