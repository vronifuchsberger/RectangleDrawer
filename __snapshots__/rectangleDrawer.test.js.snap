// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`draws a 2 by 2 rectangle 1`] = `
"
┌┐
└┘"
`;

exports[`draws a 5 by 3 rectangle 1`] = `
"
┌---┐
|   |
└---┘"
`;

exports[`draws a 5 by 5 rectangle 1`] = `
"
┌---┐
|   |
|   |
|   |
└---┘"
`;
